### PR TITLE
RIA-7213_internal_ada_appeal_decided_dismissed_document: Add Dismisse…

### PR DESCRIPTION
**Pull request by ikirsanov**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-7213

### Change description ###

- Added "Appeal decided : Dismissed - ADA (Letter)" template
- Added test - functional&unit

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by colmlynch60 — DISMISSED** [review]
LGTM

**Review by neha-aggarw — CHANGES_REQUESTED** [review]
As discussed, we need to use the existing generator to create the letter. We switch between templates based on the decisions.

**Review by colmlynch60 — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalAdaDecisionsAndReasonsDismissedLetterTemplate.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        if (respondendReviewDirectionList.isEmpty()) {
+            throw new RequiredFieldMissingException("No respondent review directions found");
+        }
+
+        if (respondendReviewDirectionList.size() > 1) {
+            throw new IllegalStateException("More than 1 respondent review direction");
+        }
+
+        Direction requestBuildCaseDirection = requestBuildCaseDirectionList.get(0);
+        Direction respondendReviewDirection = respondendReviewDirectionList.get(0);
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalAdaDecisionsAndReasonsDismissedLetterTemplate.java:76]: there is a helper method in AsylumUtils class, to get the direction due date based on the direction tag. Can you please use that instead?

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalAdaDecisionsAndReasonsDismissedLetterTemplate.java (hunk 1 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalAdaDecisionsAndReasonsDismissedLetterTemplate.java]: Done

**Review by Ryan-Nesbitt — DISMISSED** [review]

**Review by neha-aggarw — APPROVED** [review]

**Review by Ryan-Nesbitt — APPROVED** [review]

**Pull request merged**